datacontract/imports/dbt_importer.py

```diff
@@ -74,27 +78,79 @@ def import_dbt_manifest(
         dc_model = Model(
             description=model_contents.description,
             tags=model_contents.tags,
-            fields=create_fields(columns=model_contents.columns, adapter_type=adapter_type),
+            fields=create_fields(
+                manifest,
+                model_unique_id=model_contents.unique_id,
+                columns=model_contents.columns,
+                adapter_type=adapter_type,
+            ),
         )
 
         data_contract_specification.models[model_contents.name] = dc_model
 
     return data_contract_specification
 
+
 def convert_data_type_by_adapter_type(data_type: str, adapter_type: str) -> str:
     if adapter_type == "bigquery":
         return map_type_from_bigquery(data_type)
     return data_type
 
 
-def create_fields(columns: dict[str, ColumnInfo], adapter_type: str) -> dict[str, Field]:
-    fields = {
-        column.name: Field(
-            description=column.description,
-            type=convert_data_type_by_adapter_type(column.data_type, adapter_type) if column.data_type else "",
-            tags=column.tags,
+def create_fields(
+    manifest: Manifest, model_unique_id: str, columns: dict[str, ColumnInfo], adapter_type: str
+) -> dict[str, Field]:
+    fields = {column.name: create_field(manifest, model_unique_id, column, adapter_type) for column in columns.values()}
+    return fields
+
+
+def get_column_tests(manifest: Manifest, model_name: str, column_name: str) -> list[dict[str, str]]:
+    column_tests = []
+    model_node = manifest.nodes.get(model_name)
+    if not model_node:
+        raise ValueError(f"Model {model_name} not found in manifest.")
+
+    model_unique_id = model_node.unique_id
+    test_ids = manifest.child_map.get(model_unique_id, [])
+
+    for test_id in test_ids:
+        test_node = manifest.nodes.get(test_id)
+        if not test_node or test_node.resource_type != "test":
+            continue
+
+        if not isinstance(test_node, GenericTestNode):
+            continue
+
+        if test_node.column_name != column_name:
+            continue
+
+        column_tests.append(
+            {
+                "test_name": test_node.name,
+                "test_type": test_node.test_metadata.name,
+                "column": test_node.column_name,
+                "args": test_node.test_metadata.kwargs,
+            }
         )
-        for column in columns.values()
-    }
+    return column_tests
 
-    return fields
+
+def create_field(manifest: Manifest, model_unique_id: str, column: ColumnInfo, adapter_type: str) -> Field:
```

Comment:
`ColumnInfo` does not contain information about `data_tests`. Instead, `data_tests` information is contained in `Manifest`, and the `get_column_tests` function is used to determine whether the column contains a `not_null` test, and `required` is assigned.

- https://github.com/dbt-labs/dbt-core/blob/7df04b0fe461dcad3c2abcb67004178efaf445ce/core/dbt/artifacts/resources/v1/components.py#L58-L70